codechain/rpc.rs

```diff
@@ -64,6 +66,29 @@ pub fn setup_http_rpc_server(
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct IpcConfiguration {
+    pub socket_addr: String,
+}
+
+pub fn new_ipc(cfg: IpcConfiguration, deps: Arc<rpc_apis::ApiDependencies>) -> Result<IpcServer, String> {
+    let path = PathBuf::from(&cfg.socket_addr);
+    let server = setup_ipc_rpc_server(path.to_str().unwrap(), deps)?;
```

Comment:
How about passing cfg.socket_addr.as_ref() instead of the PathBuf?

I'm not sure PathBuf is necessary.

Reply:
You're right, it's not necessary. 

Lock files were automatically created and not intentionally included. I didn't know it was necessary.

I will apply your requests, and commit it again. 
Please review it again.